crates/router/src/core/payments/helpers.rs

```diff
@@ -1453,14 +1455,20 @@ pub fn make_merchant_url_with_response(
 
     let status_check = redirection_response.status;
 
-    let payment_intent_id = redirection_response.payment_id;
+    let payment_client_secret = client_secret
+        .ok_or(errors::ApiErrorResponse::InternalServerError)
+        .into_report()
+        .attach_printable("Expected client secret to be `Some`")?;
```

Comment:
@jarnura @Narayanbhat166 Should error be thrown over here as InternalServerError or client_secret_destroyed.  Currently client secret is deleted only when status is canceled or when it is expired.

Reply:
Based on offline discussion , client secret will be made mandatory so it will be removed as optional in a separate PR.

Reply:
The client secret should not be deleted, if we do not find the client secret in `PaymentAttempt` then it is an invalid state. We should throw internal server error in this case.

Reply:
* `PaymentIntent`